src/VisualStudio/CSharp/Impl/CodeModel/CSharpCodeModelService.CodeModelEventCollector.cs

```diff
@@ -928,33 +922,42 @@ protected override void EnqueueChangeEvent(SyntaxNode node, SyntaxNode parent, C
                 {
                     foreach (var variableDeclarator in ((BaseFieldDeclarationSyntax)node).Declaration.Variables)
                     {
-                        eventQueue.EnqueueChangeEvent(variableDeclarator, parent, eventType);
+                        eventQueue.EnqueueRemoveEvent(variableDeclarator, parent);
                     }
                 }
                 else if (node is AttributeListSyntax)
                 {
                     foreach (var attribute in ((AttributeListSyntax)node).Attributes)
                     {
-                        if (parent is BaseFieldDeclarationSyntax)
-                        {
-                            foreach (var variableDeclarator in ((BaseFieldDeclarationSyntax)parent).Declaration.Variables)
-                            {
-                                eventQueue.EnqueueChangeEvent(attribute, variableDeclarator, eventType);
-                            }
-                        }
-                        else
-                        {
-                            eventQueue.EnqueueChangeEvent(attribute, parent, eventType);
-                        }
+                        AddOrRemoveEventToEventQueueForAttributes(attribute, parent, eventQueue.EnqueueRemoveEvent);
                     }
                 }
+                else if (node is AttributeSyntax)
+                {
+                    AddOrRemoveEventToEventQueueForAttributes((AttributeSyntax)node, parent, eventQueue.EnqueueRemoveEvent);
+                }
                 else
                 {
-                    eventQueue.EnqueueChangeEvent(node, parent, eventType);
+                    eventQueue.EnqueueRemoveEvent(node, parent);
                 }
             }
 
-            protected override void EnqueueRemoveEvent(SyntaxNode node, SyntaxNode parent, CodeModelEventQueue eventQueue)
+            private void AddOrRemoveEventToEventQueueForAttributes(AttributeSyntax attribute, SyntaxNode parent, Action<SyntaxNode, SyntaxNode> enqueueAddOrRemoveEvent)
```

Comment:
This always adds and event that it either an add or a remove event.  Maybe just `AddEventToEventQueueForAttributes`.